### PR TITLE
#30367 Refactor job system and enhance SSE monitoring.

### LINKED ISSUE #30367: Implement Abandoned Job Detection and Recovery
### Parent Issue

https://github.com/dotCMS/core/issues/29474

### Task

We need to enhance our job queue system to handle abandoned jobs. These are jobs that may have been interrupted due to server crashes, network failures, or other unexpected issues, leaving them in an inconsistent state.

#### Objective:
Implement mechanisms to detect abandoned jobs and provide recovery strategies to ensure system reliability and data consistency.

Proposed Strategies:

1. Job Heartbeats:
   - Implement periodic heartbeat updates for running jobs
   - Create a background process to identify jobs with stale heartbeats

2. Timeout Mechanisms:
   - Add a `max_execution_time` field to job configurations
   - Implement a background process to check for jobs exceeding their maximum execution time

3. Recovery Procedures:
   - Develop a recovery process
   - Identify jobs in inconsistent states and apply appropriate recovery actions

### Additional Considerations:
- Ensure that abandoned job recovery doesn't conflict with distributed locking mechanisms
- Consider the impact on job queue performance and optimize where necessary
- Evaluate and document any changes to the system's fault tolerance and high availability characteristics

### Proposed Objective

Core Features

### Proposed Priority

Priority 2 - Important

### Acceptance Criteria

1. System can detect jobs that have been abandoned due to server crashes or other issues
2. Abandoned jobs are automatically handled according to configured recovery strategies
3. All new functionality is covered by appropriate tests
4. System performance is not significantly impacted by new abandoned job handling processes

### LINKED ISSUE #30665: Issue: Job Monitoring Stops in All Tabs When One Tab is Closed
### Parent Issue

https://github.com/dotCMS/core/issues/29498

### Problem Statement

There is an issue with the job monitoring feature. When multiple browser tabs are opened to monitor the progress of the same job, all tabs display the job progress correctly. However, if one of the tabs is closed, monitoring stops in all remaining tabs. The following error is logged when this occurs:

14:48:34.607 ERROR job.JobQueueResource - Error writing SSE event
org.apache.catalina.connector.ClientAbortException: java.io.IOException: Broken pipe
	at org.apache.catalina.connector.OutputBuffer.doFlush(OutputBuffer.java:309) ~[?:?]
	at org.apache.catalina.connector.OutputBuffer.flush(OutputBuffer.java:271) ~[?:?]
	...

### Steps to Reproduce

	1.	Open multiple browser tabs to monitor the progress of a single job.
	2.	Observe that the job progress is displayed correctly in all tabs.
	3.	Close one of the tabs monitoring the same job.
	4.	Notice that all remaining tabs stop monitoring the job, and the above error is logged in the server logs.

### Acceptance Criteria

closing one tab shouldn't stop the other 

### dotCMS Version

current main

### Proposed Objective

Technical User Experience

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by jgambarios**
Removed obsolete job events, streamlined job state management by introducing more precise states such as `FAILED_PERMANENTLY` and `ABANDONED_PERMANENTLY`. Replaced job completion terminology and refined method signatures and naming conventions to reinforce consistency. Enhanced Server-Sent Events (SSE) monitoring with a dedicated utility class for improved performance and error handling.

This PR fixes: #30367

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/jobs/business/api/JobQueueManagerAPIImpl.java [hunk 1]
```diff
@@ -360,6 +359,16 @@ public JobPaginatedResult getCompletedJobs(int page, int pageSize) throws DotDat
         }
     }
 
+    @CloseDBIfOpened
+    @Override
+    public JobPaginatedResult getSuccessfulJobs(int page, int pageSize) throws DotDataException {
```
[the remark this answers is not in the document]
Comment (jgambarios) [reply dotCMS/src/main/java/com/dotcms/jobs/business/api/JobQueueManagerAPIImpl.java]: Done

File: dotCMS/src/main/java/com/dotcms/jobs/business/api/JobQueueManagerAPIImpl.java [hunk 2]
```diff
@@ -360,6 +359,16 @@ public JobPaginatedResult getCompletedJobs(int page, int pageSize) throws DotDat
...
+        try {
+            return jobQueue.getSuccessfulJobs(page, pageSize);
```
[the remark this answers is not in the document]
Comment (jgambarios) [reply dotCMS/src/main/java/com/dotcms/jobs/business/api/JobQueueManagerAPIImpl.java:367]: Done

File: dotCMS/src/main/java/com/dotcms/jobs/business/api/events/RealTimeJobMonitor.java [hunk 3]
```diff
@@ -425,19 +421,44 @@ public boolean test(Job job) {
...
-            return job -> job.state() == JobState.FAILED
+            return job -> (job.state() == JobState.FAILED
+                    || job.state() == JobState.FAILED_PERMANENTLY)
                     && job.result().isPresent()
                     && job.result().get().errorDetail().isPresent();
         }
 
+        /**
+         * Creates a predicate that matches successful jobs. The predicate matches any job in the
+         * SUCCESS state.
```
[the remark this answers is not in the document]
Comment (jgambarios) [reply dotCMS/src/main/java/com/dotcms/jobs/business/api/events/RealTimeJobMonitor.java]: Done

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/job/JobQueueResource.java [hunk 4]
```diff
@@ -62,7 +58,7 @@ public Response createJob(
             @PathParam("queueName") String queueName,
             @BeanParam JobParams form) throws JsonProcessingException, DotDataException {
 
-        final var initDataObject = new WebResource.InitBuilder(webResource)
+        final var initDataObject = new InitBuilder(webResource)
```
[the remark this answers is not in the document]
Comment (jgambarios) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/job/JobQueueResource.java:61]: Umm not sure, I mean, to create the `InitDataObject` with  the `WebResource` it is just another way to initialize it and also the request and response it is set in the builder object: `.requestAndResponse(request, null)`, but not sure if that answered your question.

[the remark this answers is not in the document]
Comment (jgambarios) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/job/JobQueueResource.java:62]: I understand now what you meant, sorry, and yeah, I just added the response in all the `InitDataObject` objects, good suggestion, thanks!

**Review by valentinogiardino — APPROVED** [review]

**Review by fabrizzio-dotCMS — APPROVED** [review]

**Pull request merged**